interface.go

```diff
@@ -44,11 +45,13 @@ func getParams(ep *pb.Endpoint) (string, error) {
 		if len(matches) == 1 {
 			name = matches[0][1]
 			typeStr = matches[0][2]
-			params = append(params, fmt.Sprintf("%s %s", name, typeStr))
+			patternParams = append(patternParams, fmt.Sprintf("%s %s", name, typeStr))
 		} else {
-			params = append([]string{fmt.Sprintf("%s %s", name, typeStr)}, params...)
+			qp := []string{fmt.Sprintf("%s %s", name, typeStr)}
+			queryParams = append(qp, queryParams...)
```

Comment:
Do these need to be in reverse?

Reply:
Yes, I just realised I mixed up the names. So `queryParams` and `patternParams` should be swapped (updating commit pushed).
In a URL like `api/{key}/data/{start-time}` I call `key` and `startTime` patternParameters. They are listed in reverse order:  `[startTime, key]` in the protoMessage.